Shared/UI/Face Detection/FaceImageCaptureViewController.swift

```diff
@@ -17,7 +17,17 @@ import CoreImage.CIFilterBuiltins
 /// A live preview of the camera is shown on the main view.
 class FaceImageCaptureViewController: ViewController {
 
+    enum VideoCaptureState {
+        case idle
+        case starting
+        case capturing
+        case ending
+    }
```

Comment:
Implementing this as a simple state machine.

Reply:
nice. I was gonna do this too.